tests/container_tests/test_zip_container.py

```diff
@@ -97,21 +110,29 @@ def test_list(self):
         with self.fs_handler.open_as_container(self.zip_file_path) as handler:
             zip_generator = handler.list()
             zip_list = list(zip_generator)
-            self.assertIn(self.dir_path.split("/")[0], zip_list)
-            self.assertNotIn(self.zipped_file_name, zip_list)
+            self.assertIn(self.dir_name1.rstrip('/'), zip_list)
+            self.assertIn(self.dir_name2.rstrip('/'), zip_list)
+            self.assertNotIn(self.nested_zipped_file_name, zip_list)
+            self.assertNotIn(self.zipped_file_path, zip_list)
```

Comment:
One of the causes we missed this bug could be the readability of the test code. I recommend defining expected output as `set()` and compare as a set, like
```python
expected = {self.dir_name1.rstrip('/'), self.dir_name2.rstrip('/')}
zip_list = list(handler.list())
assert expected == set(zip_list)
assert len(expected) == len(zip_list)
```